### PR TITLE
Press X to open at exception in IDE

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Also add extensions thrown by the server during test execution to the
suppressed exception list, and fix multi module support for opening in the IDE.

### CODE REVIEW

**Comment by phillip-kruger** [comment]
Hi @stuartwdouglas 

I am getting 

```
2021-11-30 09:45:00,287 ERROR [io.qua.run.Quarkus] (Quarkus Main Thread) Error running Quarkus: java.lang.RuntimeException: Key e already registered
        at io.quarkus.deployment.console.ConsoleStateManager$ConsoleContext.addCommandInternal(ConsoleStateManager.java:289)
        at io.quarkus.deployment.console.ConsoleStateManager$ConsoleContext.reset(ConsoleStateManager.java:302)
        at io.quarkus.deployment.dev.IsolatedDevModeMain$1.accept(IsolatedDevModeMain.java:116)
        at io.quarkus.deployment.dev.IsolatedDevModeMain$1.accept(IsolatedDevModeMain.java:99)
        at io.quarkus.runtime.ApplicationLifecycleManager.lambda$setDefaultExitCodeHandler$0(ApplicationLifecycleManager.java:375)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:221)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

When testing with https://github.com/phillip-kruger/logging-manager-example (changing pom to 999-SNAPSHOT)

I am still investigating ...

**Comment by stuartwdouglas** [comment]
Ugh, I forgot that 'e' was already used to edit the parameters when the application exits. 

I will need to pick another letter.

**Comment by phillip-kruger** [comment]
How about x ? Also, is does not seem like netbeans is working, but that is something I can look to fix after the merge

File: core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import io.quarkus.dev.console.DevConsoleManager;
+
 public enum Ide {
 
     // see for cli syntax of idea https://www.jetbrains.com/help/idea/opening-files-from-command-line.html
-    IDEA("idea", "--line %s", "--help"),
```
Comment (Postremus) [inline core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:12]: The line parameter was used to directly point the cursor at a given line.
E.g. when clicking on a line of a stacktrace.

I remember testing this, and it worked for me.
Is there any reason to remove `--line` from the call?

**Comment by phillip-kruger** [comment]
Maybe we need to change line 288 in `io.quarkus.deployment.console.ConsoleStateManager` from 
`throw new RuntimeException("Key " + consoleCommand.getKey() + " already registered");` to
```
Holder exiting = commands.get(consoleCommand.getKey());
                    throw new RuntimeException("Key " + consoleCommand.getKey() + " with description '"
                            + exiting.consoleCommand.getDescription()
                            + "' already registered. You need to use another key for '" + consoleCommand.getDescription()
                            + "'");
```

Then the exception (we have now) is more descriptive:

```
Caused by: java.lang.RuntimeException: Key e with description 'Opens last exception in IDE' already registered. You need to use another key for 'Edits the command line parameters and restarts'
        at io.quarkus.deployment.console.ConsoleStateManager$ConsoleContext.addCommandInternal(ConsoleStateManager.java:292)
        at io.quarkus.deployment.console.ConsoleStateManager$ConsoleContext.reset(ConsoleStateManager.java:306)
        at io.quarkus.deployment.dev.IsolatedDevModeMain$1.accept(IsolatedDevModeMain.java:116)
        at io.quarkus.deployment.dev.IsolatedDevModeMain$1.accept(IsolatedDevModeMain.java:99)
        at io.quarkus.runtime.ApplicationLifecycleManager.lambda$setDefaultExitCodeHandler$0(ApplicationLifecycleManager.java:375)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:76)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        ... 6 more
```

w.d.y.t ?

File: core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java (hunk 1 above)
Comment (stuartwdouglas) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:12]: It didn't work for me, it just created an empty file called --line. 

Changing it to use the same format as the others worked for me.

**Review by phillip-kruger — APPROVED** [review]

**Comment by gsmet** [comment]
CI run looks very suspicious but I force pushed a rebase. Let's see what another run says.

**Comment by stuartwdouglas** [comment]
The wrapping of the handler caused problems for some of the tests. I have hopefully fixed it.

**Comment by gsmet** [comment]
Got confirmation in another PR that the failures are unrelated.

**Pull request merged**